fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBLuceneFileReference.java

```diff
@@ -56,12 +66,30 @@ public long getBlockSize() {
         return blockSize;
     }
 
+    public void setSegmentInfo(byte[] segmentInfo) {
+        this.segmentInfo = segmentInfo;
+    }
+
+    public void setEntries(byte[] entries) {
+        this.entries = entries;
+    }
+
+    @SpotBugsSuppressWarnings("EI_EXPOSE_REP")
+    public byte[] getSegmentInfo() {
+        return segmentInfo;
+    }
+
+    @SpotBugsSuppressWarnings("EI_EXPOSE_REP")
+    public byte[] getEntries() {
+        return entries;
+    }
+
     public Tuple getTuple() {
-        return Tuple.from(id, size, blockSize);
+        return Tuple.from(id, size, blockSize, segmentInfo, entries);
```

Comment:
I think maybe this has reached the point where we ought to switch to Protobuf.

It's only going to get worse when we add the compression algorithm, encryption IV, and so on.

Reply:
Good point.  You want another radar for it or to address in this one @MMcM?

Reply:
Changed to Protobuf.

Reply:
Thanks @tian-yizuo